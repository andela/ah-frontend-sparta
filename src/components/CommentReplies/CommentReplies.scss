@import '../../assets/scss/shared.scss';

.commentDate {
  font-size: rem(12);
}
 .replyCard-header {
    background-color: transparent;
 }
