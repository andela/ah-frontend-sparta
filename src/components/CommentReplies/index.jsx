import React from 'react';
import moment from 'moment';
import './CommentReplies.scss';
import EditCommentDialog from '../EditCommentDialog';

const CommentReplies = (
  {
    reply: {
      id, body, author, createdAt,
    },
    slug,
    deleteComment,
    updateComment,
    onChangeComment,
  },
) => (
  <div>
    <div className="card commectCard offset-1">
      <div className="card-header commentHeader replyCard-header">
        <div className="authorDataContainer">
          {author.image ? (
            <img
              className="imagePlaceholder"
              src={author.image}
              alt="Commenter"
            />
          ) : (
            <div className="imagePlaceholder" />
          )}
          <div className="authorData">
            <b>{author.username}</b>
            <div className="text-muted commentDate">
              {moment(createdAt.slice(0, 10)).format('LL')}
            </div>
          </div>
        </div>
        <div className="dropdown commentMenu">
          <button
            className="btn dropdown-toggle"
            type="button"
            id="dropdownMenuButton"
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="false"
          />
          <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
            <button
              type="button"
              className="dropdown-item"
              data-toggle="modal"
              data-target="#exampleModalCenter"
            >
              Edit
            </button>
            <button
              id="deleteButton"
              type="button"
              className="dropdown-item"
              onClick={() => deleteComment(slug, id)}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
      <div className="card-body">{body}</div>
    </div>
    <EditCommentDialog comment={body} updateComment={updateComment} onChangeComment={onChangeComment} id={id} />
  </div>
);

export default CommentReplies;
