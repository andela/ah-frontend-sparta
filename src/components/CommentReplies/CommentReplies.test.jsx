import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CommentReplies from './index';

test('CommentReplies snapshot test', () => {
  const props = {
    reply: {
      id: 1,
      body: 'Hi',
      author: 'Rogha',
      createdAt: '12:35',
    },
    slug: 'comment-slug',
    deleteComment: jest.fn(),
    updateComment: jest.fn(),
    onChangeComment: jest.fn(),
  };
  const wrapper = shallow(<CommentReplies {...props} />);
  wrapper.find('#deleteButton').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
