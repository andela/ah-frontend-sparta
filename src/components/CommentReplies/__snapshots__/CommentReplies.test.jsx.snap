// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentReplies snapshot test 1`] = `ShallowWrapper {}`;
